src/main/java/com/google/sps/servlets/QueryServlet.java

```diff
@@ -0,0 +1,85 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+
+import java.io.IOException;
+
+import com.google.sps.data.Response;
+import com.google.sps.data.Restaurant;
+import com.google.sps.data.User;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.List;
+import java.util.ArrayList;
+
+@WebServlet("/query")
+public class QueryServlet extends HttpServlet {
+
+    private final String apiKey = "AIzaSyBL_9GfCUu7DGDvHdtlM8CaAywE2bVFVJc";
+    private final Gson gson = new GsonBuilder().create();
+    private User user;
+
+    @Override
+    public void doGet(HttpServletRequest servletRequest, HttpServletResponse servletResponse) throws IOException {
+        HttpSession session = servletRequest.getSession(false);
+        Response response = (Response) session.getAttribute("response");
+        if(response == null)
+            servletResponse.getWriter().println(gson.toJson(new Response("NO_RESULTS", null)));
+        else if (response.getStatus().equals("OK"))
+            response.pick();
+        servletResponse.getWriter().println(gson.toJson(response));
+    }
+
+    @Override
+    public void doPost(HttpServletRequest servletRequest, HttpServletResponse servletResponse) throws IOException, ServletException {
+        String lat = servletRequest.getParameter("lat");
+        String lon = servletRequest.getParameter("lng");
+        String radius = servletRequest.getParameter("radius");
+        String type = "restaurant";
+        String searchTerms = servletRequest.getParameter("searchTerms");
+        String urlStr = "https://maps.googleapis.com/maps/api/place/nearbysearch/json?location=" + lat + "," + lon + "&radius=" + radius + "&type=" + type + "&keyword=" + searchTerms + "&key=" + apiKey;
+
+        URLConnection conn = new URL(urlStr).openConnection();
+        conn.connect();
+
+        JsonElement jsonElement = new JsonParser().parse(new InputStreamReader(conn.getInputStream()));
+        JsonObject responseJson = jsonElement.getAsJsonObject();
+        Response response = gson.fromJson(responseJson, Response.class);
+
+        HttpSession session = servletRequest.getSession(true);
+        if (response.getStatus().equals("OK"))
+            response.pick();
+        session.setAttribute("response", response);
+        session.setAttribute("user", new User(Integer.parseInt(servletRequest.getParameter("priceLevel"))));
+        servletResponse.getWriter().println(gson.toJson(response));
+        servletRequest.getRequestDispatcher("/searches").forward(servletRequest, servletResponse);
```

Comment:
this works for now. Let's add a TODO to factor out the datastore interaction into a helper library/class to make it explicit what we're persisting. We don't need the entire HttpServletRequest structure in order to persist to datastore; we only need the request parameters, and that could just be some method.

Reply:
done